drivers/SPI.cpp

```diff
@@ -115,29 +162,29 @@ int SPI::write(const char *tx_buffer, int tx_length, char *rx_buffer, int rx_len
 {
     lock();
     _acquire();
-    int ret = spi_master_block_write(&_spi, tx_buffer, tx_length, rx_buffer, rx_length, _write_fill);
+    int ret = spi_transfer(&_self->spi, tx_buffer, tx_length, rx_buffer, rx_length, &_write_fill);
```

Comment:
Again, this use of _write_fill assumes little endian. Possibly simplest to have a 3-way `uint8_t`/`uint16_t`/`uint32_t` union to store it.